File: Part1_강의자료/Ch05_정렬/문제별코드/1302_베스트셀러/1302_map.py
N = int(input())
titles = {}

for i in range(N):
    title = input().strip()
    if title not in titles:
        titles[title] = 1
    else:
        titles[title] += 1

max_title = None
max_cnt = 0

for title, cnt in titles.items():
    if cnt > max_cnt or (cnt == max_cnt and title < max_title):
        max_title = title
        max_cnt = cnt

print(max_title)

File: Part1_강의자료/Ch05_정렬/문제별코드/1302_베스트셀러/1302_map.cpp
#include <stdio.h>
#include <map>
#include <string>
using namespace std;

int main()
{
    int N;
    map<string, int> titles;
    char title[51];

    scanf("%d", &N);
    for (int i = 0; i < N; i++)
    {
        scanf("%s", title);
        titles[title]++;
    }

    string max_title = "";
    int max_cnt = 0;
    for (auto title : titles)
    {
        if (title.second > max_cnt || (title.second == max_cnt && title.first < max_title))
        {
            max_title = title.first;
            max_cnt = title.second;
        }
    }

    printf("%s\n", max_title.c_str());
    return 0;
}
